### DISCUSSION TITLE
How do I open q-drawer from another component in Quasar V2?

### DISCUSSION THREAD

**Discussion by MandingaRadi** [quasarframework/quasar#8931] [General - Components / Directives / etc] [answered]
In Quasar V1 I did the following:

IN MAIN LAYOUT:
I have a q-drawer with a v-Model called "Drawer".

In Created() (Main Layout):
 this.$root.$on ('Drawer', this.openDrawer)

In Methods (Main Layout):
openDrawer () {this.Drawer =! this.Drawer}

IN ANOTHER COMPONENT:
In Methods (Another Component):
this.$root.$emit('Drawer')

It worked fine in Quasar V1, but the problem is that "$on" no longer works in Vue 3.
Sorry for my English, and thank you!

**Comment by metalsadman** [comment, marked as answer]
If you still rely on this pattern, then read vue docs migration about it here https://v3.vuejs.org/guide/migration/events-api.html#overview.

**Reply by MandingaRadi** [reply]
Metalsadman, Thanks for answering!
I decided to do it with Vuex (I already had it built in) and it works wonders.

Whoever has the same problem, I can gladly share my code.

Thanks again!
